### PR TITLE
Grab json-pushes for multiple pushes at a time when iterating over children/parents

### LINKED ISSUE #443: Batch calls to hgmo when detecting regressions
Sarah found out that we are spending a lot of time querying `hg.mozilla.org` when detecting regressions and pointed out that we can gather data for much data at a time:
https://hg.mozilla.org/mozilla-central/json-pushes?version=2&startID=38350&endID=38355

Since we know how many pushes we will look into the future or past (MAX_DEPTH), we should be able to grab all the `hg.m.o` data we need in a single query.

### PR DESCRIPTION
**Pull request by ahal**
Fixes #443

### CODE REVIEW

File: mozci/util/hgmo.py [hunk 1]
```diff
@@ -25,7 +26,8 @@ class HGMO:
     )
 
     # instance cache
-    CACHE: Dict[Tuple[str, str], HGMO] = {}
+    CACHE: Dict[Tuple[str, str], HgRev] = LRU(1000)
+    JSON_PUSHES_CACHE: Dict[int, Dict[str, Any]] = LRU(1000)
```
Comment (marco-c) [inline mozci/util/hgmo.py:30]: Nit: You could define a Push object like: `HgPush = NewType("HgPush", Dict[str, Any])`

File: mozci/util/hgmo.py [hunk 2]
```diff
@@ -36,20 +38,66 @@ def __init__(self, rev, branch="autoland"):
...
-        HGMO.CACHE[key] = instance
+        if key in HgRev.CACHE:
+            return HgRev.CACHE[key]
+        instance = HgRev(rev, branch)
+        HgRev.CACHE[key] = instance
         return instance
 
-    def _get_resource(self, url):
+    @staticmethod
+    def _get_and_cache_pushes(branch, url):
```
Comment (marco-c) [inline mozci/util/hgmo.py:48]: Nit: can you add type information to the new functions you added? It should be simple in these cases

**Review by marco-c — APPROVED** [review]

**Comment by marco-c** [comment]
@ahal could you tag a new release after merging this?

File: mozci/util/hgmo.py (hunk 2 above)
Comment (ahal) [reply mozci/util/hgmo.py:48]: Oops, yes I need to get in the habit more.

**Pull request merged**